tests/rbd_mirror/test_rbd_mirror_daemon_restart_imagesync.py

```diff
@@ -0,0 +1,159 @@
+    )
+    ec_image_spec = (
+        f'{config["ec_pool_config"]["pool"]}/{config["ec_pool_config"]["image"]}'
+    )
+
+    def initiate_resync_and_wait_sync_to_start(img_spec):
+        mirror2.resync(img_spec)
+        time.sleep(5)
+        mirror2.wait_for_status(
+            imagespec=img_spec, state_pattern="down+unknown", retry_interval=5
+        )
+
+    def verify_resync(img_spec):
+        """Observe sync percentage till image goes to up+replying"""
+        sync_percent = 0
+        while mirror2.mirror_status("image", img_spec, "state") == "up+syncing":
+            curr_sync_percent = int(
+                mirror2.mirror_status("image", img_spec, "description")[-3:-1]
+            )
+            if curr_sync_percent >= sync_percent:
+                sync_percent = curr_sync_percent
+                continue
+            log.info(
+                f"img_spec current sync percent:{curr_sync_percent} is greater than or eaqual to previous sync_percent"
+            )
+            if curr_sync_percent < sync_percent:
+                log.err("Syncing seemed to be restarted, test case failed")
+                return 1
+            time.sleep(2)
+        return 0
+
+    def restart_daemon_till_sync_completes(img_spec):
+        """Untill cync of images in img_spec completes and restart count is less than 11,
+        restart rbd mirror daemon"""
+        restart_count = 0
+        while (
+            (mirror2.mirror_status("image", img_spec[0], "state") != "up+replaying")
+            and (mirror2.mirror_status("image", img_spec[1], "state") != "up+replaying")
+            and restart_count < 10
+        ):
+            log.info(f"Restarting rbd-mirror daemon {restart_count}st/nd/rd/th time")
+            mirror2.change_service_state(None, "restart")
+            time.sleep(60)
+            restart_count += 1
+
+        log.info(
+            "Restarted mirror daemon till image sync completed or restarted 10 times during execution, method returns 0"
+        )
+        return 0
+
+    decoy_imagespec = f'{config["rep_pool_config"]["pool"]}/decoy_'
+
+    def decoy_setup():
+        # Set decoy - increase number of images under mirroring and decrease concurrent imagesyncs
+        # to get confirtable window for cephci to monitor images under focus for verification
+        mirror2.exec_cmd(
+            cmd="ceph config set global rbd_mirror_concurrent_image_syncs 1"
+        )
+        for i in range(1, 10):
+            mirror2.create_image(imagespec=decoy_imagespec + f"{i}", size="1G")
+            mirror2.image_feature_enable(
+                imagespec=decoy_imagespec + f"{i}", image_feature="journaling"
+            )
+
+    def generate_background_ios():
+        for i in range(1, 3):
+            for i in range(1, 10):
+                mirror1.benchwrite(imagespec=decoy_imagespec + f"{i}", io="10M")
+            time.sleep(5)
+        return 0
+
+    with parallel() as p:
```

Comment:
In the block that you are pointing out, we are adding data into images so that when we execute next steps, we would be having sufficient size in image to observe the image operation, 

If we merge parallel blocks, we would not be able to make sure that we have sufficient data filled in before we resync the image.
resync without sufficient data, doesn't make sense.

Reply:
ok got the point, Thanks!